Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1541 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_NEXUSGEOMETRY_NEXUSGEOMETRYSAVETEST_H_
+#define MANTID_NEXUSGEOMETRY_NEXUSGEOMETRYSAVETEST_H_
+
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/ComponentInfoBankHelpers.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidGeometry/Instrument/InstrumentVisitor.h"
+#include "MantidKernel/EigenConversionHelpers.h"
+#include "MantidKernel/ProgressBase.h"
+#include "MantidKernel/WarningSuppressions.h"
+#include "MantidNexusGeometry/NexusGeometryDefinitions.h"
+#include "MantidNexusGeometry/NexusGeometrySave.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+
+#include <H5Cpp.h>
+#include <algorithm>
+#include <boost/filesystem.hpp>
+#include <boost/make_shared.hpp>
+#include <cxxtest/TestSuite.h>
+#include <fstream>
+#include <gmock/gmock.h>
+#include <iostream>
+#include <memory>
+#include <set>
+
+using namespace Mantid::NexusGeometry;
+
+//---------------------------------------------------------------
+namespace {
+
+const std::string DEFAULT_ROOT_PATH = "raw_data_1";
+using FullH5Path = std::vector<std::string>;
+
+// get path as string. Used for the dependency tests.
+std::string toH5PathString(FullH5Path &path) {
+  std::string pathString = "";
+  for (const std::string &grp : path) {
+    pathString += "/" + grp;
+  }
+  return pathString;
+}
+
+class MockProgressBase : public Mantid::Kernel::ProgressBase {
+public:
+  GNU_DIAG_OFF_SUGGEST_OVERRIDE
+  MOCK_METHOD1(doReport, void(const std::string &));
+  GNU_DIAG_OFF_SUGGEST_OVERRIDE
+};
+
+//  local Class used for validation of the structure of a nexus file as needed
+//  for the unit tests.
+class HDF5FileTestUtility {
```

Comment:
I think we should put the `HDF5FileTestUtility` and the `ScopedFile` classes in their own header files. Can live in the `test` directory. Otherwise they are taking up quite a lot of room here, and I think they could be useful elsewhere.

Reply:
done